sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -954,7 +953,7 @@ private async Task RunAsync(CancellationToken cancellationToken)
                     {
                         if (!ActivePartitionProcessors.TryGetValue(kvp.Key, out var activeTaskAndTokenSource) || activeTaskAndTokenSource.Item1.IsCompleted)
                         {
-                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.Shutdown, cancellationToken).ConfigureAwait(false);
+                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.OwnershipLost, cancellationToken).ConfigureAwait(false);
```

Comment:
Updated this because, as far as I know, the whole point of the stopped reason was to tell the users whether they could checkpoint or not (Shutdown = safe to checkpoint, OwnershipLost = not safe). They would need to hold a reference to the last processed `ProcessEventArgs`, though.

We are stopping the processing here because an error occurred, so we can assume it's not safe to checkpoint.

Reply:
This has always been a bit confusing to me, so let me know if I got something wrong.